### PR TITLE
Delete LectureHalls

### LINKED ISSUE #223: Delete Lecture halls

### PR DESCRIPTION
**Pull request by mono424**
I added the global "Delete"-Function, to send delete requests to the backend. If not wanted, this could be changed to a post request with a different URL for example "lectureHalls/:id/delete" but I liked delete request more :)

### CODE REVIEW

File: web/template/admin/admin_tabs/lecture_halls.gohtml [hunk 1]
```diff
@@ -21,7 +21,12 @@
             pwrCtrlIp: '{{$lectureHall.PwrCtrlIp}}',
             id: '{{$lectureHall.ID}}',}"
              class="w-full border-2 dark:border-secondary border-gray-200 rounded mb-6">
-            <div class="dark:bg-secondary bg-gray-100 font-thin text-xl text-3 p-2 px-6">{{$lectureHall.Name}}</div>
+            <div class="dark:bg-secondary bg-gray-100 font-thin text-xl text-3 p-2 px-6">
+                {{$lectureHall.Name}}
+                <i class="fas fa-trash hover:text-1 px-2"
+                   title="Delete Lecture"
```
Comment (joschahenningsen) [inline web/template/admin/admin_tabs/lecture_halls.gohtml]:
```suggestion
                   title="Delete Lecture Hall"
```

File: dao/lecture_halls.go [hunk 2]
```diff
@@ -45,6 +45,10 @@ func GetLectureHallByID(id uint) (model.LectureHall, error) {
 	return lectureHall, err
 }
 
+func DeleteLectureHall(id uint) error {
+	return DB.Delete(&model.LectureHall{}, id).Error
```
Comment (joschahenningsen) [inline dao/lecture_halls.go]: What happens with streams that are scheduled for the lecture hall we're about to delete? IMO, streams.lecture_hall_id should be set to `null` here. e.g. using DB.Raw. Also, camera presets for the lecture hall could be deleted

**Review by joschahenningsen — CHANGES_REQUESTED** [review]
Looks good @mono424  :) 
Just a quick thought.

File: dao/lecture_halls.go (hunk 2 above)
Comment (mono424) [reply dao/lecture_halls.go]: true! thanks for the catch! 👍

Comment (mono424) [reply dao/lecture_halls.go]: @joschahenningsen I think this should be fine now, used `DB.Exec` instead of `DB.Raw` :)

**Review by joschahenningsen — APPROVED** [review]
Looks good, thanks! :)

**Pull request merged**